### ISSUE TITLE
"Raspberry Pi 3B+ Not Booting with OP-TEE Image on SD Card"

### ISSUE THREAD

**Issue by Chae0510** [OP-TEE/optee_os#6442]
Following the instructions provided at [optee.readthedocs.io](https://optee.readthedocs.io/en/latest/building/devices/rpi3.html) I successfully copied the img file to my SD card without any issues. However, when I insert this SD card into my Raspberry Pi 3B+ board and power it on, the ACT LED blinks briefly three times, and the board doesn't boot.

I've also tried following the steps in the link [OP-TEE developer build v3.22.0 fails to boot on new RPi 3B+ board](https://github.com/OP-TEE/optee_os/issues/6284#issue-1883113001), as someone with a similar device faced the same issue, but I still couldn't get the SD card to be read.

I'm using a Raspberry Pi 3 Model B+ (2017 version), and I'm wondering if anyone has encountered a similar error and found a solution. I've tried all the solutions I could find on Google without success.

I'll be waiting for any advice or suggestions. Thank you in advance.

**Comment by nolzaheo** [comment]
I tried these steps and it worked on rpi3 b+

$ apt update && apt upgrade -y
$ apt install -y     adb     acpica-tools     autoconf     automake     bc     bison     build-essential     ccache     cpio     cscope     curl     device-tree-compiler     e2tools     expect     fastboot     flex     ftp-upload     gdisk     git     libattr1-dev     libcap-dev     libfdt-dev     libftdi-dev     libglib2.0-dev     libgmp3-dev     libhidapi-dev     libmpc-dev     libncurses5-dev     libpixman-1-dev     libslirp-dev     libssl-dev     libtool     libusb-1.0-0-dev     make     mtools     netcat     ninja-build     python3-cryptography     python3-pip     python3-pyelftools     python3-serial     python-is-python3     rsync     swig     unzip     uuid-dev     wget     xdg-utils     xterm     xz-utils     zlib1g-dev
$ mkdir -p test
$ cd test
$ curl https://storage.googleapis.com/git-repo-downloads/repo > /bin/repo && chmod a+x /bin/repo
$ repo init -u https://github.com/OP-TEE/manifest.git -m rpi3.xml
$ repo sync -j4 --no-clone-bundle
$ cd build
$ make -j2 toolchains
$ make -j `nproc`
$ make sdcard-image

**Comment by jjsshh3116** [comment]
@nolzaheo 
I try this time but failed...
Can you provide specific information such as build environment and Raspberry Pi board version?

**Comment by jbech-linaro** [comment]
I'm the one maintaining the RPi3 build for OP-TEE, last time I tried was at the 4.0.0 release. Back then [it was working as expected](https://github.com/OP-TEE/optee_os/pull/6341#pullrequestreview-1687824736). Unfortunately, we don't have the RPi3 on a CI loop, but I can give it a try again to see whether there is a new regression not seen before.

**Comment by jjsshh3116** [comment]
@jbech-linaro Thank you for your response! I currently have both the RPi3 B and RPi3 B+ boards. When I use the same SD card image (OP-TEE 4.0.0), it boots up normally only on the RPi3 B board. Therefore, with an experimental approach, I am planning to try creating a boot image for the OP-TEE on the B+ board using buildroot.

**Comment by jbech-linaro** [comment]
I've been testing a bit this morning and I suspect there is some issues with the dependency graph in the Makefile, since building it seems to sometimes not complete the build and rebuilding solves it. I'll see if I can figure something out.

**Comment by jbech-linaro** [comment]
I've made some changes in https://github.com/OP-TEE/build/pull/706 which I believe solves an issue that the build sometimes not complete as it should, basically some dependency error in the Makefile. Regarding the version, I also the "Rpi3 B v1.2 from 2016".



> When I use the same SD card image (OP-TEE 4.0.0), it boots up normally only on the RPi3 B board.

I do think I can check the B+. But, I will cross check if I have a B+ laying around, since I know I have used a B+ in the past as well I think.
